vyper/semantics/analysis/module.py

```diff
@@ -66,6 +67,29 @@ def __init__(
 
         # TODO: Move computation out of constructor
         module_nodes = module_node.body.copy()
+        const_var_decls = [
+            n for n in module_nodes if isinstance(n, vy_ast.VariableDecl) and n.is_constant
+        ]
+
+        while const_var_decls:
```

Comment:
this kind of long algorithmic thing in the constructor jumps out at me. if it's a really important routine, it can get a helper function